dataset/src/main/scala/frameless/ops/GroupByOps.scala

```diff
@@ -173,4 +185,52 @@ class GroupedBy2Ops[K1, K2, V](
   def flatMapGroups[U: TypedEncoder](f: ((K1, K2), Iterator[V]) => TraversableOnce[U]): TypedDataset[U] = {
     underlying.flatMapGroups(f)
   }
+
+  def pivot[P: CatalystPivotable, Values <: HList](pivotColumn: TypedColumn[V, P],
+                                values: Values)(
+    implicit validValues: MatchesAll[Values, P]
+  ): Pivot[V, TypedColumn[V,K1] :: TypedColumn[V, K2] :: HNil, P, Values] =
+    Pivot(self, g1 :: g2 :: HNil, pivotColumn, values)
 }
+
+/** Represents a typed Pivot operation.
+  */
+final case class Pivot[T, GroupedColumns <: HList, PivotType, Values <: HList]
+(ds: TypedDataset[T],
+ groupedBy: GroupedColumns,
+ pivotedBy: TypedColumn[T, PivotType],
+ values: Values) {
+
+  object agg extends ProductArgs {
+    def applyProduct[
+    AggrColumns <: HList,
+    AggrColumnTypes <: HList,
+    GroupedColumnTypes <: HList,
+    NumValues <: Nat,
+    TypesForPivotedValues <: HList,
+    TypesForPivotedValuesOpt <: HList,
+    OutAsHList <: HList,
+    Out](aggrColumns: AggrColumns)(
+      implicit
+      tc: AggregateTypes.Aux[T, AggrColumns, AggrColumnTypes],
+      tl1: ToList[GroupedColumns, Any],
+      tl2: ToList[AggrColumns, Any],
+      tl3: ToList[Values, Any],
```

Comment:
I wouldn't ask for a type class to do this, just implement it by by hand: (hopefully it both faster to compiler and to run!)

    def mapAny[X](h: HList)(f: Any => X): List[X] =
      h match {
        case HNil    => Nil
        case x :: xs => f(x) :: mapAny(xs)(f)
      }